### PR TITLE
Logs app: Add index pattern content

### PR DESCRIPTION
**Pull request by EamonnTP**
### Summary

This PR adds content regarding Kibana index patterns that are available for selection in the Logs app settings. 

### Docs preview 

https://observability-docs_725.docs-preview.app.elstc.co/guide/en/observability/master/configure-data-sources.html#edit-config-settings

### CODE REVIEW

File: docs/en/observability/configure-logs-sources.asciidoc [hunk 1]
```diff
@@ -23,7 +23,11 @@ default configuration settings.
 
 | *Name* | Name of the source configuration. 
 
-| *Indices* | Index pattern or patterns in the {es} indices to read log data from.
+| *Index pattern* | Index pattern or patterns in the {es} indices to read log data from.
+
+Each log source now integrates with {kib} index patterns which support creating and
```
Comment (bmorelli25) [inline docs/en/observability/configure-logs-sources.asciidoc:28]: What are *Kibana index patterns*? This term was added to our "What's New" doc as well, but I can't find any mention of it in the [runtime fields docs](https://www.elastic.co/guide/en/kibana/7.13/managing-index-patterns.html), or anywhere else in the Kibana guide.

**Comment by EamonnTP** [comment]
@bmorelli25 I think the term _Kibana Index Patterns_ relates to the standard index patterns described [here](https://www.elastic.co/guide/en/kibana/current/index-patterns.html). Previously, the Logs UI used Elasticsearch index name patterns. Now the user has a choice to configure a Kibana index pattern instead to incorporate [runtime fields](https://www.elastic.co/guide/en/kibana/current/managing-index-patterns.html#runtime-fields), which is not available with Elasticsearch index name patterns.

Hi @weltenwort Could you help clarify this is little more? Thanks!

**Comment by weltenwort** [comment]
@EamonnTP thanks for taking care of the docs :heart: Your explanation is absolutely on point. We're using the prefix "Kibana" to "index patterns" to differentiate them from the index name patterns (strings such as `filebeat-*`) which the ES docs also call index patterns in some places. The Kibana team is aiming to avoid that ambiguity by renaming Kibana index patterns to "data views" in the next major version: elastic/kibana#94244.

As for the Logs UI, using Kibana index patterns (KIP) will become the preferred style of configuration soon. The user will be prompted to choose a KIP when they visit the Logs UI for the first time. And we'll incorporate a flyout that allows for creation of new KIPs right there without leaving the app.

**Comment by EamonnTP** [comment]
Thanks @weltenwort! Is this doc update sufficient enough or do you feel we need to add more clarification?

**Review by bmorelli25 — APPROVED** [review]
Fantastic. Appreciate the explanation @EamonnTP and @weltenwort!

**Comment by weltenwort** [comment]
@EamonnTP how about adding something like

> You can continue to use log sources configured to use an index name pattern such as `filebeat-*` instead of a Kibana index pattern, but some features like those depending on runtime fields might not be available.

Would that be useful info for the user?

**Comment by EamonnTP** [comment]
> @EamonnTP how about adding something like
> 
> > You can continue to use log sources configured to use an index name pattern such as `filebeat-*` instead of a Kibana index pattern, but some features like those depending on runtime fields might not be available.
> 
> Would that be useful info for the user?

Thanks @weltenwort I added the additional context.

**Pull request merged**